packages/core/src/env-set/create-pool-by-env.ts

```diff
@@ -1,25 +1,48 @@
-import {
-  assertEnv,
-  conditional,
-  conditionalString,
-  getEnv,
-  Optional,
-} from '@silverhand/essentials';
+import { assertEnv, conditional, getEnv, Optional } from '@silverhand/essentials';
 import inquirer from 'inquirer';
 import { createPool } from 'slonik';
 import { createInterceptors } from 'slonik-interceptor-preset';
+import { z } from 'zod';
 
-import { createDatabase, createDatabaseCli } from '@/database/seed';
+import { createDatabase, createDatabaseCli, replaceDsnDatabase } from '@/database/seed';
 
 import { appendDotEnv } from './dot-env';
 import { allYes, noInquiry } from './parameters';
 
 const defaultDatabaseUrl = getEnv('DB_URL_DEFAULT', 'postgres://@localhost:5432');
 const defaultDatabaseName = 'logto';
 
+const createDatabaseIfNotExists = async (dsn: string): Promise<[string, boolean]> => {
```

Comment:
maybe `initDatabase` is a more accurate name?

since the function `createDatabaseIfNotExists` also deals with the existing database.

Reply:
y